### PR TITLE
[JENKINS-42228] - Prevent JavaScript error on the Jenkins system configuration page

### PR DESCRIPTION
**Pull request by takashiharano**
See [JENKINS-42228](https://issues.jenkins-ci.org/browse/JENKINS-42228).

This is a workaround for TypeError by handling undefined object on the Jenkins system configuration page.

The error still occurs in Jenkins 2.250.
Manage Jenkins > Configure Ststem

    behavior.js:119 TypeError: Cannot read property 'firstChild' of undefined
        at registerValidator (hudson-behavior.js:511)
        at Array.forEach (<anonymous>)
        at behavior.js:116
        at Array.forEach (<anonymous>)
        at behavior.js:111
        at Array.forEach (<anonymous>)
        at Object.applySubtree (behavior.js:93)
        at select.js:280

The permanent solution is probably to fix the HTML structure or something.
I tried to investigate the root cause, but it seems to be a timing problem and a little bit hard to figure out.
Even so, this workaround would not take a change for the worse.

Here is the error details for further investigation.

error_description

### Proposed changelog entries

* Entry 1: Prevent JavaScript error when registering validators in some cases

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: war/src/main/webapp/scripts/hudson-behavior.js [hunk 1]
```diff
@@ -500,7 +500,11 @@ var tooltip;
 //========================================================
 // using tag names in CSS selector makes the processing faster
 function registerValidator(e) {
-    e.targetElement = findFollowingTR(e, "validation-error-area").firstChild.nextSibling;
+    var tr = findFollowingTR(e, "validation-error-area");
+    if (!tr || !tr.firstChild) {
```
Comment (timja) [inline war/src/main/webapp/scripts/hudson-behavior.js:504]: can we add a warning?
```suggestion
    if (!tr || !tr.firstChild) {
      console.warn("Couldn't register validator, start element was", e);
```

Comment (takashiharano) [reply war/src/main/webapp/scripts/hudson-behavior.js:504]: Sure, I added a warning log.
The output is like this:
warning

**Review by timja — APPROVED** [review]
Change looks good to me I probably wouldn’t call it a violation.

And would reword the changelog too.

Prevent JavaScript error when registering validators in some cases?

**Comment by takashiharano** [comment]
@timja 
Thank you for reviewing.
I also updated the changelog.

**Review by oleg-nenashev — APPROVED** [review]
Looks good as a workaround 👍 
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Review by fqueiruga — APPROVED** [review]

**Review by jetersen — APPROVED** [review]
LGTM, thanks for catching the bug 😅

**Comment by oleg-nenashev** [comment]
Thanks a lot for you contributions @takashiharano !
